### PR TITLE
Lex dev

### PR DESCRIPTION
**Pull request by ross-desmond**
Updates lex-ros1 to use lex-common and provide full code coverage.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lex_node/src/lex_node.cpp [hunk 1]
```diff
@@ -12,199 +12,68 @@
...
+  ros_response.text_response = lex_response.text_response;
+  ros_response.slots = std::vector<lex_common_msgs::KeyValue>();
+  std::transform(lex_response.slots.begin(), lex_response.slots.end(),
+                 std::back_inserter(ros_response.slots), [](const std::pair<std::string, std::string>& slot) {
+              lex_common_msgs::KeyValue key_value;
+              key_value.key = slot.first;
+              key_value.value = slot.second;
+              return key_value;
+          });
+  return ros_response;
```
Comment (mm318) [inline lex_node/src/lex_node.cpp]: Is there a difference between `AudioTextConversation::Response` and `lex_common_msgs::AudioTextConversationResponse`? Be consistent by sticking to one type.

Comment (timrobotson) [reply lex_node/src/lex_node.cpp]: I'm surprised this works. I thought the return value and first argument had to be the same type.

File: lex_node/src/lex_node.cpp [hunk 2]
```diff
@@ -12,199 +12,68 @@
...
 
 bool LexNode::LexServerCallback(lex_common_msgs::AudioTextConversationRequest & request,
                                 lex_common_msgs::AudioTextConversationResponse & response)
 {
-  if (!lex_runtime_client_) {
-    // @todo define a new exception
-    AWS_LOG_WARN(__func__, "Lex runtime client is not initialized, LoadConfiguration.");
-    throw std::invalid_argument("Lex runtime client is not initialized, LoadConfiguration.");
+  LexRequest lex_request;
+  lex_request << request;
```
Comment (timrobotson) [inline lex_node/src/lex_node.cpp]: Why do operator overloading rather than just a `convert_ros_lex_message_to_lex_request` function?

Comment (ross-desmond) [reply lex_node/src/lex_node.cpp]: No reason really, pretty much identical.

File: lex_node/src/lex_node.cpp (hunk 1 above)
Comment (ross-desmond) [reply lex_node/src/lex_node.cpp]: There is no difference between the two, they are the same type. I will change it to be consistent.

**Comment by nburek** [comment]
This PR is blocked on the lex-common package being bloom released.

**Pull request closed without merging**